### ISSUE TITLE
Regex literal within a control flow statement (i.e. `guard`, `if`, etc.)

### ISSUE THREAD

**Issue by jjgp** [nicklockwood/SwiftFormat#1287]
Given an input such as this:

```Swift
guard urlRequest.url?.scheme?.wholeMatch(of: /http|https/) != nil else {
    return nil
}
```

The output after formatting is the following (notice the space between the `|` in the regex literal):
```Swift
guard urlRequest.url?.scheme?.wholeMatch(of: /http | https/) != nil else {
    return nil
}
```

I would expect the output to be without the added spaces.

I have reproduced the issues in a few unit tests on this [branch](https://github.com/jjgp/SwiftFormat/blob/583e1037c39991452a019e21c65deb598b891ff1/Tests/FormatterTests.swift#L464). I am using Swift 5.7 and will double check the options for setting the versions.

**Comment by nicklockwood** [comment]
Thanks for reporting - I can confirm this is a bug.

**Comment by nicklockwood** [comment]
@jjgp fixed in 0.50.2
